typing/subst.ml

```diff
@@ -470,6 +471,33 @@ and modtype_declaration s decl  =
     mtd_loc = loc s decl.mtd_loc;
   }
 
+(* Same as [signature] except than instead of making the items local, we rescope
```

Comment:
I think it would be cleaner to make this function internal to `Env` and then create an `Env.enter_signature` function that calls `Env.add_signature` on the result of this function. This function doesn't use anything in `Subst` and isn't really related to the other functions in it. The `enter_` vs. `add_` distinction would be consistent with that used for things like `add_module`.

Reply:
I'm kind of torn of that one: on principal I agree with you, in practice it means duplicating a non-zero amount of code that is already present in subst.

I'll push one last commit doing the change and we can decide then if we want to include it or not.

Reply:
Done in commit 047a986 ; as I said, I don't really like it but I defer the choice to you.

Btw, I contest:
> This function doesn't use anything in `Subst`

It used an internal function of `Subst` (which in turns uses a lot of functions of `Subst`), which makes no sense out of context, so I chose to duplicate it instead of exporting it.

Reply:
I hadn't noticed that `signature_component` was not merely applying a substitution to a signature component. However, you could easily adjust things so that it was. Just make `rename_bound_idents` and `refresh_bound_idents` change the bound identifiers as they go (i.e. have them return a substitution and a new signature instead of a substitution and a list of identifiers). Then `signature_component` is just like all the other functions in `Subst` and can reasonably be exposed in the mli and used in `Env`.

Reply:
Ah indeed. I updated the commit.